code-challenges.test.js

```diff
@@ -15,24 +15,76 @@
 // a) Create a test with an expect statement using the variable provided.
 // HINT: Check out this resource: https://jestjs.io/docs/expect#expectarraycontainingarray
 
+describe("shuffle", () => {
+    it("takes in an array, removes the first item from the array, and shuffles the remaining items", () => {
+      expect(shuffle(["purple", "blue", "green", "yellow", "pink"])).toEqual(expect.arrayContaining(["yellow", "blue", "pink", "green"]))
+      expect(shuffle(["chartreuse", "indigo", "periwinkle", "ochre", "aquamarine", "saffron"])).toEqual(expect.arrayContaining(["saffron", "aquamarine", "periwinkle", "indigo", "ochre"]))
+    })
+  })
+
+
 const colors1 = ["purple", "blue", "green", "yellow", "pink"]
 // Expected output example (can be a different order): ["yellow", "blue", "pink", "green"]
+
+
 const colors2 = ["chartreuse", "indigo", "periwinkle", "ochre", "aquamarine", "saffron"]
 // Expected output example (can be a different order): ["saffron", "aquamarine", "periwinkle", "indigo", "ochre"]
 
+
+
 // b) Create the function that makes the test pass.
 
+const shuffle = (arr) => {
+    if (arr.length > 0) {
+      const firstItem = arr.shift()
+      
+      for (let i = arr.length - 1; i > 0; i--) {
+        const shuff = Math.floor(Math.random() * (i + 1)) 
+        [arr[i], arr[shuff]] = [arr[shuff], arr[i]]
+      }
+      
+      arr.unshift(firstItem)
+    }
+    
+    return arr
+  }
```

Comment:
Logic looks great!  Some things to consider in the future, what is your test coming back with?  Is it passing, or are you running into errors?  I also like to check back with the prompt and make sure I am solving the same problem (ex. removing the first element in the array, you don't need to add it back in).